presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -10,39 +16,38 @@ class AnonymizerEngine:
     and replaces the PII entities with the desired transformations.
     """
 
-    # Task: 2672
-    # TODO this needs to be implemented currently a stab.
-    # TODO replace analyze_results with domain results
-    # Notice the document Omri created, it impacts the implementation
     def __init__(
             self,
-            text: str,
-            # TODO change to domain object
-            transformations: List[str],
-            # TODO change to domain object
-            analyze_results: List[str],
+            data: dict,
     ):
-        """
-        Handle text replacement for PIIs with requested transformations.
+        """Handle text replacement for PIIs with requested transformations.
 
-        :param text: The original text we want to replace PIIs in
-        :param transformations: The desired transformations - mapping between PII type
-        and transformation type with relevant params
-        :param analyze_results: The results of the analyzer of PIIs locations and scores
+        :param data: a map which contains the transformations, analyzer_results and text
         """
-        self.transformations = transformations
-        self.analyze_results = analyze_results
-        self.text = text
-        self._end_point = len(text)
+        self._transformations = Transformations(data.get("transformations"))
+        self._analyze_results = data.get("analyzer_results")
+        if self._analyze_results is None:
+            self._analyze_results = AnalyzerResults()
+        self._text = data.get("text")
+        if self._text is None:
+            self._text = ""
+        self._end_point = len(self._text)
 
     def anonymize(self):
         """Anonymize method to anonymize the given text.
 
         :return: the anonymized text
         """
-        # TODO a loop that goes through the analyzer results from END to START! reverse.
-        # TODO for each result, replace the old value with the new value
-        # Make sure we handle partial intersections using the endpoint param
-        # TODO dictionary with the transformations and fields
-        # To map field to its transformation
-        pass
+        analyzer_results = self._analyze_results.to_sorted_set(True)
+        for result in analyzer_results:
+            transformation = self._transformations.get_transformation(result)
+            anonymizer_class = anonymizers.get(transformation.get("type").lower())
```

Comment:
I would have a map between the type and the class, less hacky, and will also raise a more specifc exception in case a non-existent type is passed

Reply:
The mapping between the type and the class is a map.

Reply:
right, my bad.